x/autopilot/module_ibc.go

```diff
@@ -196,10 +196,14 @@ func (im IBCModule) OnRecvPacket(
 		}
 		im.keeper.Logger(ctx).Info(fmt.Sprintf("Forwaring packet from %s to stakeibc", newData.Sender))
 
-		// Try to liquid stake - return an ack error if it fails, otherwise return the ack generated from the earlier packet propogation
-		if err := im.keeper.TryLiquidStaking(ctx, packet, newData, routingInfo); err != nil {
-			im.keeper.Logger(ctx).Error(fmt.Sprintf("Error liquid staking packet from autopilot for %s: %s", newData.Sender, err.Error()))
-			return channeltypes.NewErrorAcknowledgement(err)
+		switch routingInfo.Action {
+		case types.LiquidStake:
+			// Try to liquid stake - return an ack error if it fails, otherwise return the ack generated from the earlier packet propogation
+			if err := im.keeper.TryLiquidStaking(ctx, packet, newData, routingInfo); err != nil {
+				im.keeper.Logger(ctx).Error(fmt.Sprintf("Error liquid staking packet from autopilot for %s: %s", newData.Sender, err.Error()))
+				return channeltypes.NewErrorAcknowledgement(err)
```

Comment:
do we know if partial state changes get reverted when an ack error is thrown?

Reply:
confirmed it does [[ref](https://github.com/cosmos/ibc-go/blob/v7.1.0/modules/core/keeper/msg_server.go#L433-L458)] - marking resolved